### PR TITLE
Redefine Kingdom system services to support v2alpha public API.

### LINKED ISSUE #53: Redefine Kingdom system services to support v2alpha API.
This covers redefining the resources and services. Updating the implementation for these services will be tracked separately.

### PR DESCRIPTION
**Pull request by SanjayVas**
Closes #53. This is intended to be merged into the main branch after the base PR #44 has been merged.

---
This change is [Reviewable](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement/55)

### CODE REVIEW

**Review by wangyaopw — APPROVED** [review]
I get the idea that in resource oriented design we want a separate API for each resource. But I personally really don't like so many resources. I feel all these are just dealing with the computation resource. The log and participant etc are just property of a computation.

If all methods are in the same ComputationsService, we only need one CoroutineStub in the client, whose code looks much nicer.

No strong opinion anyway.

**Review by SanjayVas — COMMENTED** [review]
Smaller resources and therefore more focused services are preferred. This is following similar principles that one would follow in object-oriented design, with the key difference being that resources always have IDs/keys.

e.g. in a theoretical OO design: `computation.getParticipant(key)` returns a `ComputationParticipant` object with a `setRequisitionParams()` method.

> *[Reviewable](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement/55#-:-M_M66jC-SUiujpUqQmJ:b-k51gfx)* status: 0 of 7 files reviewed, all discussions resolved

**Review by efoxepstein — APPROVED** [review]
> Reviewed 7 of 7 files at r1.
> *[Reviewable](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement/55)* status: :shipit: complete! all files reviewed, all discussions resolved (waiting on @SanjayVas)

**Review by stevenwarejones — APPROVED** [review]
same thoughts as @wangyaopw but lgtm

> *[Reviewable](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement/55#-:-M_MyzpF3zD4a9MgesQP:brj91wr)* status: :shipit: complete! all files reviewed, all discussions resolved (waiting on @kungfucraig)

File: src/main/proto/wfa/measurement/system/v1alpha/computation_log_entries_service.proto [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+      returns (ComputationLogEntry);
+}
+
+// Request message for the `CreateComputationLogEntry` method.
+message CreateComputationLogEntryRequest {
+  // Resource key of the parent `ComputationParticipant`. Required.
+  ComputationParticipant.Key parent = 1;
+
+  // The `ComputationLogEntry` to create. Required. The `key` field will be
+  // ignored, and the system will assign an ID.
```
[the remark this answers is not in the document]
Comment (zachcwc) [reply src/main/proto/wfa/measurement/system/v1alpha/computation_log_entries_service.proto:40]: Ah, thought more and I guess the key is unrelated to the actual computation that needs to be created

**Review by SanjayVas — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement/55)* status: all files reviewed, 1 unresolved discussion (waiting on @zachcwc)

___
*[src/main/proto/wfa/measurement/system/v1alpha/computation_log_entries_service.proto, line 40 at r1](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement/55#-M_8Mid8TCBtXcoXjtYO-r1-40:-M_R5STi3uAIM3HTXwJs:b-tckyz7) ([raw file](https://github.com/world-federation-of-advertisers/cross-media-measurement/blob/f0965392bc0e2f6edc72be71497170683d6fdaa6/src/main/proto/wfa/measurement/system/v1alpha/computation_log_entries_service.proto#L40)):*
<details><summary><i>Previously, zachcwc wrote…</i></summary><blockquote>

Ah, thought more and I guess the key is unrelated to the actual computation that needs to be created
</blockquote></details>

The `key` field in each resource hold the resource key, which contains the ID of the resource and those of its ancestors. Due to the fact that we have the system generate IDs rather than having the user specify them, the `key` field is generally output-only.

For a [`Create` standard method](https://google.aip.dev/133) like this one, the request message has a `parent` field with the key of the parent resource, and the system will generate the ID of the new resource.

See the [Design section](https://github.com/world-federation-of-advertisers/cross-media-measurement-api#design) of the API README for where our API design deviates from the AIPs.

**Pull request merged**